From ab99ad19897ab5db5281d907898d75b95c31f79a Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Tue, 28 Jan 2025 14:45:00 +0100
Subject: [PATCH 1569/1695] dts: stm32: align EXTI nodes to upstream bindings
 on stm32mp131

Use the same binding merged in upstream Linux kernel v6.10

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1b9cc51c2d547c5087c48a4664c2fbdf60af8562 dts: stm32: align EXTI nodes to upstream bindings on stm32mp131]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I28cbb8b447e2b7953df569a3472ed9bd480f6eea
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434916
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi | 119 ++++++++++++++++++------------
 1 file changed, 72 insertions(+), 47 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index b57a865b4..0fdc0ceed 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -302,54 +302,79 @@
 			compatible = "st,stm32mp1-exti";
 			interrupt-controller;
 			#interrupt-cells = <2>;
-			#address-cells = <0>;
 			reg = <0x5000d000 0x400>;
-
-			exti-interrupt-map {
-				#address-cells = <0>;
-				#interrupt-cells = <2>;
-				interrupt-map-mask = <0xffffffff 0>;
-				interrupt-map =
-					<0  0 &intc GIC_SPI 6   IRQ_TYPE_LEVEL_HIGH>,
-					<1  0 &intc GIC_SPI 7   IRQ_TYPE_LEVEL_HIGH>,
-					<2  0 &intc GIC_SPI 8   IRQ_TYPE_LEVEL_HIGH>,
-					<3  0 &intc GIC_SPI 9   IRQ_TYPE_LEVEL_HIGH>,
-					<4  0 &intc GIC_SPI 10  IRQ_TYPE_LEVEL_HIGH>,
-					<5  0 &intc GIC_SPI 24  IRQ_TYPE_LEVEL_HIGH>,
-					<6  0 &intc GIC_SPI 65  IRQ_TYPE_LEVEL_HIGH>,
-					<7  0 &intc GIC_SPI 66  IRQ_TYPE_LEVEL_HIGH>,
-					<8  0 &intc GIC_SPI 67  IRQ_TYPE_LEVEL_HIGH>,
-					<9  0 &intc GIC_SPI 68  IRQ_TYPE_LEVEL_HIGH>,
-					<10 0 &intc GIC_SPI 41  IRQ_TYPE_LEVEL_HIGH>,
-					<11 0 &intc GIC_SPI 43  IRQ_TYPE_LEVEL_HIGH>,
-					<12 0 &intc GIC_SPI 77  IRQ_TYPE_LEVEL_HIGH>,
-					<13 0 &intc GIC_SPI 78  IRQ_TYPE_LEVEL_HIGH>,
-					<14 0 &intc GIC_SPI 106 IRQ_TYPE_LEVEL_HIGH>,
-					<15 0 &intc GIC_SPI 109 IRQ_TYPE_LEVEL_HIGH>,
-					<16 0 &intc GIC_SPI 1   IRQ_TYPE_LEVEL_HIGH>,
-					<19 0 &intc GIC_SPI 3   IRQ_TYPE_LEVEL_HIGH>,
-					<21 0 &intc GIC_SPI 32  IRQ_TYPE_LEVEL_HIGH>,
-					<22 0 &intc GIC_SPI 34  IRQ_TYPE_LEVEL_HIGH>,
-					<23 0 &intc GIC_SPI 73  IRQ_TYPE_LEVEL_HIGH>,
-					<24 0 &intc GIC_SPI 93  IRQ_TYPE_LEVEL_HIGH>,
-					<25 0 &intc GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>,
-					<26 0 &intc GIC_SPI 38  IRQ_TYPE_LEVEL_HIGH>,
-					<27 0 &intc GIC_SPI 39  IRQ_TYPE_LEVEL_HIGH>,
-					<28 0 &intc GIC_SPI 40  IRQ_TYPE_LEVEL_HIGH>,
-					<29 0 &intc GIC_SPI 72  IRQ_TYPE_LEVEL_HIGH>,
-					<30 0 &intc GIC_SPI 53  IRQ_TYPE_LEVEL_HIGH>,
-					<31 0 &intc GIC_SPI 54  IRQ_TYPE_LEVEL_HIGH>,
-					<32 0 &intc GIC_SPI 83  IRQ_TYPE_LEVEL_HIGH>,
-					<33 0 &intc GIC_SPI 84  IRQ_TYPE_LEVEL_HIGH>,
-					<44 0 &intc GIC_SPI 96  IRQ_TYPE_LEVEL_HIGH>,
-					<47 0 &intc GIC_SPI 92  IRQ_TYPE_LEVEL_HIGH>,
-					<48 0 &intc GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>,
-					<50 0 &intc GIC_SPI 117 IRQ_TYPE_LEVEL_HIGH>,
-					<52 0 &intc GIC_SPI 118 IRQ_TYPE_LEVEL_HIGH>,
-					<53 0 &intc GIC_SPI 119 IRQ_TYPE_LEVEL_HIGH>,
-					<68 0 &intc GIC_SPI 63  IRQ_TYPE_LEVEL_HIGH>,
-					<70 0 &intc GIC_SPI 98  IRQ_TYPE_LEVEL_HIGH>;
-			};
+			interrupts-extended =
+				<&intc GIC_SPI 6   IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_0 */
+				<&intc GIC_SPI 7   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 8   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 9   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 10  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 24  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 65  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 66  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 67  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 68  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 41  IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_10 */
+				<&intc GIC_SPI 43  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 77  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 78  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 106 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 109 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 1   IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<&intc GIC_SPI 128 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 3   IRQ_TYPE_LEVEL_HIGH>,
+				<0>,						/* EXTI_20 */
+				<&intc GIC_SPI 32  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 34  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 73  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 93  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 38  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 39  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 40  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 72  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 53  IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_30 */
+				<&intc GIC_SPI 54  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 83  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 84  IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,						/* EXTI_40 */
+				<0>,
+				<&intc GIC_SPI 75  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 75  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 96  IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 92  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<&intc GIC_SPI 117 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_50 */
+				<0>,
+				<&intc GIC_SPI 118 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 119 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,						/* EXTI_60 */
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 63  IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<&intc GIC_SPI 98  IRQ_TYPE_LEVEL_HIGH>;	/* EXTI_70 */
 		};
 
 		syscfg: syscon@50020000 {
-- 
2.47.2

